### PR TITLE
Add building depth imposters for TS base structures.

### PR DESCRIPTION
**Pull request by pchote**
This takes us another small step towards enabling the new depth rendering in TS.

When testing you can quickly toggle between normal and depth rendering using the <kbd>`</kbd> key.

### CODE REVIEW

**Comment by pchote** [comment]
Note that this sets `EnableDepthBuffer: True` in mod.yaml, which breaks rendering of many things (including voxels, lasers, sprites being drawn partially under the terrain).  This PR is fixing one corner of that problem space.

File: OpenRA.Game/Widgets/RootWidget.cs [hunk 1]
```diff
@@ -47,6 +48,10 @@ public override bool HandleKeyPress(KeyInput e)
 
 					return true;
 				}
+
+				var world = Game.OrderManager.World;
+				if (world != null && world.LocalPlayer != null && hk == new Hotkey(Keycode.BACKQUOTE, Modifiers.None))
```
Comment (Mailaender) [inline OpenRA.Game/Widgets/RootWidget.cs]: People with non-US keyboards can use `new Hotkey(Keycode.D, Modifiers.Ctrl)` instead.

File: OpenRA.Mods.Common/Traits/Render/WithGateSpriteBody.cs [hunk 2]
```diff
@@ -26,6 +26,9 @@ class WithGateSpriteBodyInfo : WithSpriteBodyInfo, IWallConnectorInfo, Requires<
 		[Desc("Wall type for connections")]
 		public readonly string Type = "wall";
 
+		[Desc("Override sequence to use when fully open (if non-null).")]
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithGateSpriteBody.cs]: The `(if non-null)` isn't obvious when looking at just the traits documentation. Just skip it.

**Comment by Mailaender** [comment]
This does something. :white_check_mark:  I have no idea if it is correct though.

**Comment by Phrohdoh** [comment]
c038c36f901e269999c440cb95147af70fb9cc0a does more than what the commit message says, though that is one of the uses.

These look good to me. The east and west-most corners of structures still have some issues but I think those would need custom depth sprites to look any better. 👍

I couldn't convince you to do the same for RA2, could I?

**Comment by GraionDilach** [comment]
That depth offset looks like something which can be added to the building art importer to me, atleast as a rough estimate by duplicating the second offset.

Why does it need a new sprite though and why can't buildngz.shp be used?

Also why did the Wolverine got additional z-offset as well, but the Titan/deployed MSA/tick tanks didn't?

**Comment by pchote** [comment]
* Depth importing will be part of a later pr that sets up the civilian actors.
* TS manages its depth buffer differently, and its buildngz.shp doesn't work for us.
* The wolverine is there in the testcase commit to help test depth sorting only.  This PR isn't about vehicles (which will remain completely broken until voxel rendering is fixed)

**Comment by obrakmann** [comment]
:+1:, I guess. Please remove the test-case.

**Pull request merged**